### PR TITLE
Creating Building Class

### PR DESCRIPTION
**Pull request by rickyc**
**Description**
Scaffolded a few classes as described below.

**Class(es)**
* Building
* Company
* Floor

**Implementation Notes**
Buildings contain many floors. Each floor contains one or multiple companies.

**Outstanding Tasks**
No

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -3,6 +3,12 @@
   <head>
     <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.1.0/jquery.min.js"></script>
 
+    <!-- Include Models -->
+    <script src="scripts/building.js"></script>
+     <script src="scripts/floor.js"></script>
+     <script src="scripts/company.js"></script>
+ <script src="scripts/application.js"></script>
```
Comment (rickyc) [inline index.html]: Fix indentation! 😡

File: scripts/application.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+$(document).ready(() => {
+ var building = new Building("598 Broadway", "598 Broadway");
```
Comment (rickyc) [inline scripts/application.js]: Fix indentation!!! It should be `const` 😠.

File: scripts/building.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    this.floors[x] = floor;
+    return floor;
+  }
+
+toString() {
+  console.log(`${this.name} is located at ${this.address}.`);
+  console.log(`This building has ${this.floors.length - 1} floors constructed.`);
+
+    for (var i = 0; i < this.floors.length ; i ++) {
+          var f= this.floors[i]
```
Comment (thebluecat5445) [inline scripts/building.js]: FIX THE INDENTATION

File: scripts/application.js [hunk 4]
```diff
@@ -0,0 +1,8 @@
...
+var floor = building.constructFloor(15);
+var thoughtWorks = new Company("ThoughtWorks");
```
Comment (Noahb930) [inline scripts/application.js]: should be const

File: scripts/application.js (hunk 2 above)
Comment (Marioisdabest) [inline scripts/application.js]: ya

File: scripts/building.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
+class Building {
```
Comment (Noahb930) [inline scripts/building.js]: @rickyc setup houndci-bot

File: scripts/company.js [hunk 6]
```diff
@@ -0,0 +1,6 @@
+class Company {
+constructor(n, desc= "") {
```
Comment (thebluecat5445) [inline scripts/company.js]: fix indentation

File: scripts/building.js [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+  constructor(name,address) {
+   this.name = name;
+    this.address = address;
+     this.floors = [];
+    }
+
+  constructFloor(x) {
+    var floor = new Floor(x);
```
Comment (Noahb930) [inline scripts/building.js]: var should be const

File: scripts/building.js [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+    this.floors[x] = floor;
+    return floor;
+  }
+
+toString() {
+  console.log(`${this.name} is located at ${this.address}.`);
+  console.log(`This building has ${this.floors.length - 1} floors constructed.`);
+
+    for (var i = 0; i < this.floors.length ; i ++) {
```
Comment (Noahb930) [inline scripts/building.js]: var should be let

File: scripts/building.js (hunk 3 above)
Comment (thebluecat5445) [inline scripts/building.js]: var should be const

File: scripts/application.js [hunk 9]
```diff
@@ -0,0 +1,8 @@
+$(document).ready(() => {
+ var building = new Building("598 Broadway", "598 Broadway");
+var floor = building.constructFloor(15);
```
Comment (Marioisdabest) [inline scripts/application.js]: const again

File: scripts/application.js (hunk 4 above)
Comment (Marioisdabest) [inline scripts/application.js]: const again

**Comment by rickyc** [comment]
@thebluecat5445 @Noahb930 @Marioisdabest Thanks for the feedback! I think I've addressed it :). Can you take a look at my pull request and review it? Thanks!

File: scripts/building.js [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+    const floor = new Floor(number);
+    this.floors[number] = floor;
+    return floor;
+  }
+
+  toString() {
+    console.log(`${this.name} is located at ${this.address}.`);
+    console.log(`This building has ${this.floors.length - 1} floors constructed.`);
+
+    for (let floor of this.floors) {
```
Comment (rickyc) [inline scripts/building.js]: This should be a `const`.

File: scripts/building.js [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+      if (floor) {
+        console.log(`Floor: ${floor.number}`);
+        for (let company of floor.companies) {
```
Comment (rickyc) [inline scripts/building.js]: This should also be a `const`.

**Pull request merged**